lib/puppet/provider/package/pkgin.rb

```diff
@@ -12,8 +12,8 @@
   def self.parse_pkgin_line(package)
 
     # e.g.
-    #   vim-7.2.446 =        Vim editor (vi clone) without GUI
-    match, name, version, status = *package.match(/(\S+)-(\S+)(?: (=|>|<))?\s+.+$/)
+    #   vim-7.2.446;Vim editor (vi clone) without GUI
+    match, name, version, status = *package.match(/(\S+)-(\S+?);(?:(=|>|<))?.+$/)
```

Comment:
Oh yes, even on redhat ;-)

I'm starting to wonder if it's actually making sense to add the backwards compat. SmartOS (another consumer of pkgin) has pkgin > 0.7.0 and given the sorry state of providers on NetBSD I doubt many people (if anyone) has used them somewhat recently...